### ISSUE TITLE
[Proposal] Updates, fixes and new features

### ISSUE THREAD

**Issue by jenkin** [webhintio/markdown-link-validator#28]
I think this project has unique features (ie. built-in globbing and direct md analysis without compilation to html) among competitors (ie. [markdown-link-check](https://github.com/tcort/markdown-link-check)) and is good enough to deserve a proper evolution path in next future.

For an [internal open source project](https://github.com/sparkfabrik/company-playbook) (a knowledge base written in markdown and powered by [Raneto](https://raneto.com/)) I took the v1.0.1 (8 Jun 2021) of markdown-link-validator and made some changes to adapt it to our requirements: https://github.com/jenkin/markdown-link-validator/.

1. update all dependencies and resolve all known vulnerabilities highlighted by `npm audit`
2. add support to absolute local links (ie. `[internal page](/path/to/page.md)`)
3. [`-q` option] add a quiet mode to report only errors if present (see issue #6)
4. [`-e` option] add support to optional `.md` extension (ie. for `[internal page](./path/to/page)` it searches also for `[internal page](./path/to/page.md)` or `[internal page](./path/to/page/index.md)`
5. [`-o` option] add support to other extension checking (ie. `[local asset](./path/to/image.png)` (see issue #27)
6. ignore links in code blocks and inline code
7. add support to standalone links checking (ie. without `[anchor](https...)` or `[anchor] https...`)
8. [`-c` option] ignoring failing external urls based on response status codes (added to final report)
9. [`--ignorePatternsFrom` option] ignoring files by patterns taken from a file
10. [`--noEmptyFiles` option] raising error if linked files are empty

No breaking changes to the API are introduced. All existing options have the same behavior and defaults and new optional features depends on options disabled by default. Only 2), 6) and 7) are opinionated and could require some discussions. All new features have tests (but I know we can improved them further).

All changes live on [feature/sparkfabrik-enhancements](https://github.com/jenkin/markdown-link-validator/tree/feature/sparkfabrik-enhancements) branch. I think all of them can met a broader interest beyond our company playbook, so I propose you to review my work and consider to accept a PR and update the npm package.

**Comment by molant** [comment]
@jenkin nice!
Do you mind opening a PR so it's easier to see the diff?

**Comment by mi-hol** [comment]
> made some changes to adapt it to our requirements

the list of addl. requirements makes perfect sense!

**Comment by jenkin** [comment]
Cool! My roadmap for coming days includes two more steps:

10. Check and fix #7 
11. Add _warnings_ (ie. shown in report in yellow, but no error code on exit) for empty md files and ignored status codes (see 8)

That would be all for the purpose of our internal project.

Please, let me know if something needs to change to merge the PR (ie. I added `dist/` folder to repo to allow npm to install the package from github).

**Comment by jenkin** [comment]
@molant I finished to implement steps 10 and 11. Are you going to consider to merge my PR? If so, let's talk to finalize it! :)

**Comment by molant** [comment]
Yes, I started looking at it a few days back but job got in the way and I’m about to go on vacation for a few weeks with no laptop.
It will be a bit until I have time to finalize reviewing it.
Thanks for your patience 🙏
________________________________

**Comment by jenkin** [comment]
No problem, I will be on vacation for the next two weeks, so take your time. Thanks!

**Comment by jenkin** [comment]
Hi @molant , I'm back, I hope you are well too. Let me know next steps to close this issue... 😃

**Comment by mi-hol** [comment]
> It will be a bit until I have time to finalize reviewing it.

@molant hope you are doing fine. Would you be in a position to forecast the rough timeline(s) until completion?
